### PR TITLE
Create output directory if non-existent

### PR DESCRIPTION
**Pull request by jceplaras**
#### :rocket: Why this change?

Filed in https://github.com/apiaryio/dredd/issues/899. Create directories if non-existent.

#### :memo: Related issues and Pull Requests

Resolved https://github.com/apiaryio/dredd/issues/899

#### :white_check_mark: What didn't I forget?

- [x] To write docs
- [x] To write tests
- [x] To put [Conventional Changelog](https://dredd.readthedocs.io/en/latest/contributing/#sem-rel) prefixes in front of all my commits and run `npm run lint`

### CODE REVIEW

File: test/unit/reporters/x-unit-reporter-test.coffee [hunk 1]
```diff
@@ -65,10 +66,11 @@ describe 'XUnitReporter', () ->
     afterEach () ->
       fsStub.appendFileSync.restore()
 
-    it 'should write opening to file', (done) ->
+    it.only 'should write opening to file', (done) ->
```
Comment (honzajavorek) [inline test/unit/reporters/x-unit-reporter-test.coffee]: Forgotten `.only` 🙂

Comment (jceplaras) [reply test/unit/reporters/x-unit-reporter-test.coffee]: Will fix! Sorry about that. :)

**Comment by jceplaras** [comment]
@honzajavorek Already addressed requested changes and also added an integration test for a non-existent directory.

**Comment by honzajavorek** [comment]
I'm afraid we have a little blocker here: https://github.com/substack/node-mkdirp/pull/138

**Comment by honzajavorek** [comment]
I'm sorry I suggested `mkdirp` in the beginning. I should have inspect the quality of the package first and not to suggest the first thing npm search offered me.

I think [make-dir](https://github.com/sindresorhus/make-dir) should be fine, [fs-extra](https://github.com/jprichardson/node-fs-extra) as well. The latter includes many common utilities we could use in Dredd in other places as well - especially in the `./scripts`, which we would like to eventually keep in form of JavaScript rather than bash, for Windows compatibility. @michalholasek which one would you prefer? Do you think `fs-extra` is a smarter choice in this case? I've seen you already using `rimraf` somewhere.

**Comment by michalholasek** [comment]
@honzajavorek Both look good to me, but [fs-extra](https://github.com/jprichardson/node-fs-extra) is more "futureproof" (featurewise) for us.

**Pull request closed without merging**